app/backend/src/core/clients/blockchain/starknet/StarknetClient.ts

```diff
@@ -0,0 +1,123 @@
+import { Config } from "@/config";
+import Logger from "@/tools/Logger";
+import {
+  Block,
+  IBlockchainClient,
+  Log,
+  TransactionReceipt,
+} from "@/core/clients/blockchain/IBlockchainClient";
+import { constants, getChecksumAddress, hash, RpcProvider } from "starknet";
+import { TransferLogEvent } from "@/core/controllers/appraiser/handlers/BaseHandler";
+
+class StarknetClient implements IBlockchainClient {
+  private readonly logger: Logger;
+  private readonly provider: RpcProvider;
+
+  public readonly chainId: number;
+
+  private readonly TransferEventHash = hash.getSelectorFromName("Transfer");
+
+  constructor(config: Config, logger: Logger) {
+    this.logger = logger;
+    this.chainId = config.starknetModule.chainId;
+
+    const headers = config.indexers.starknetApiKey
+      ? { "x-apikey": config.indexers.starknetApiKey }
+      : undefined;
+
+    this.provider = new RpcProvider({
+      nodeUrl: config.indexers.starknetRpcEndpoint,
+      headers: headers,
+      chainId: constants.StarknetChainId.SN_MAIN,
+    });
+  }
+
+  public async getCurrentHeight(): Promise<number> {
+    return await this.provider.getBlockNumber();
+  }
+
+  public async getBlock(
+    blockHeight: number | string,
+  ): Promise<Block | undefined> {
+    const block = await this.provider.getBlockWithTxs(blockHeight);
+
+    if (!("block_hash" in block)) {
+      this.logger.error(
+        `Pending block parsing is not supported: ${blockHeight}`,
+      );
+      return undefined;
+    }
+
+    return {
+      hash: block.block_hash,
+      number: 1,
+      timestamp: block.timestamp,
+      // block has 2 gas prices: for ETH and for STRK
+      // both are handled as Transfer events to the sequencer address in the logs
+      baseFeePerGas: 0n,
```

Comment:
While for now we agreed it's ok to handle fee payment as part of the `Transfer` logs, there's another issue I noticed:
- For swap transactions we take max of all transfers, which on Starknet include fee. It's almost always one of the smallest ones, so we are basically ignoring it.
- For other chains, all fee payments are added separately and are always included in the total block value.

I can add a TODO for Starknet to fix this in the wrapping-up week, assuming there will be one. Gas paid should not have any significant impact on the resulting charts, especially after switch to blobs.